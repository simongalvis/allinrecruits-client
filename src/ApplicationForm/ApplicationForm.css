form{
    display: flex;
    flex-direction: column;
}
input{
    align-self: center;
}