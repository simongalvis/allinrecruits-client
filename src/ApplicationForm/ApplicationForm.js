import React from 'react'
import ApiContext from '../ApiContext';
import './ApplicationForm.css'


class ApplicationForm extends React.Component{
static contextType = ApiContext;

    handleSubmit = e =>{
        e.preventDefault();
        console.log('I am working')
        const { applicationFullname, applicationPhonenumber, applicationEmail, applicationPosition, applicationResume } = e.target;
        
        
        const submission = {
            fullname: applicationFullname.value,
            phonenumber: applicationPhonenumber.value,
            email: applicationEmail.value,
            interestedposition: applicationPosition.value,
            resumelink: applicationResume.value
        }

        this.context.addSubmission(submission);

        console.log(submission)
    }
     openPositions =['French teacher','Spanish teacher', 'Robotics instructor', 'Art instructor']
     state ={
         counter: 0
     }


    render(){
        return(
            <div className="ApplicationForm">
                <form onSubmit={this.handleSubmit}>
                    <label htmlFor="applicationFullname"  className="required"><b>Full name:</b></label>
                    <input type="text" id="applicationFullname" name="applicationFullname" placeholder="John Smith" autoCorrect="off" autoCapitalize="none" required/>
                    <label htmlFor="applicationPhonenumber"><b>Phone number:</b></label>
                    <input type="tel" id="applicationPhonenumber" name="applicationPhonenumber" placeholder="123-456-7890" autoCorrect="off" autoCapitalize="none" />
                    <label htmlFor="application-email"><b>Email:</b></label>
                    <input type="email" id="applicationEmail" name="applicationEmail" placeholder="johnsmith123@email.com" autoCorrect="off" autoCapitalize="none" />
                    {/* <label htmlFor="interestedPositionCheckboxList"><b>Interested Position(s):</b></label> */}
                    {/* Interested position checkbox list */}
                    {/* <div id="interestedPositionCheckboxList">
                        

                        <div className="checkboxLabelAndInput">
                            <input type="checkbox" id="subject1" className="checkboxInput" name="subject1" value="Spanish teacher"/>
                            <label htmlFor="subject1" className="checkboxLabel">Spanish teacher</label>
                        </div>
                        
                        <div className="checkboxLabelAndInput">
                            <input type="checkbox" id="subject2" className="checkboxInput" name="subject2" value="French teacher"/>
                            <label htmlFor="subject2" className="checkboxLabel">French teacher</label>
                        </div>
                        <div className="checkboxLabelAndInput">
                            <input type="checkbox" id="subject3" className="checkboxInput" name="subject3" value="Robotics instructor"/>
                            <label htmlFor="subject3" className="checkboxLabel">Robotics instructor</label>
                        </div>
                        <div className="checkboxLabelAndInput">
                            <input type="checkbox" id="subject4" className="checkboxInput" name="subject4" value="Art instructor"/>
                            <label htmlFor="subject4" className="checkboxLabel">Art instructor</label>
                        </div>
                        
                    </div> */}
                    <label htmlFor="applicationPosition"><b>Interested Position(s):</b></label>

                    <select name="applicationPosition" id="applicationPosition">
                        <option value="Spanish teacher">Spanish teacher</option>
                        <option value="French teacher">French teacher</option>
                        <option value="Robotics instructor">Robotics instructor</option>
                        <option value="Art instructor">Art instructor</option>
                    </select>
                    <label htmlFor="applicationResume" className="required"><b>Resume</b>(Link)</label>
                    <input type="url" id="applicationResume" name="applicationResume" placeholder="https://drive.google.com/file/d/examplelink3648384748/view" autoCorrect="off" autoCapitalize="none" required/>
                    {/* <label htmlFor="application-resume" className="required"><b>Resume</b>(pdf)</label>
                    <input type="file" id="application-resume" name="application-resume" autoCorrect="off" autoCapitalize="none" required/> */}
                    <button id="application-submit-btn" type="submit"><b>Apply</b></button>
                </form>
                
                
            </div>
        )
    }
}

export default ApplicationForm;