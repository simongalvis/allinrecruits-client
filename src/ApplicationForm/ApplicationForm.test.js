import React from 'react'
import ReactDOM from 'react-dom'
import ApplicationForm from './ApplicationForm'
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'

describe(`ApplicationForm component`, () =>{
    it(`renders without crashing`, () =>{
        const div = document.createElement('div');

        ReactDOM.render(<BrowserRouter><ApplicationForm/></BrowserRouter>, div)

        ReactDOM.unmountComponentAtNode(div);
    })
    it(`Renders UI as expected`, () =>{
        const tree = renderer
            .create(<BrowserRouter><ApplicationForm/></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})