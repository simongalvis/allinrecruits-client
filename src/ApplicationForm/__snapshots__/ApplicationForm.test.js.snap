// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationForm component Renders UI as expected 1`] = `
<div
  className="ApplicationForm"
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="required"
      htmlFor="applicationFullname"
    >
      Full name:
    </label>
    <br />
    <input
      autoCapitalize="none"
      autoCorrect="off"
      id="applicationFullname"
      name="applicationFullname"
      placeholder="John Smith"
      required={true}
      type="text"
    />
    <br />
    <label
      htmlFor="applicationPhonenumber"
    >
      Phone number:
    </label>
    <br />
    <input
      autoCapitalize="none"
      autoCorrect="off"
      id="applicationPhonenumber"
      name="applicationPhonenumber"
      placeholder="123-456-7890"
      type="tel"
    />
    <br />
    <label
      htmlFor="application-email"
    >
      Email:
    </label>
    <br />
    <input
      autoCapitalize="none"
      autoCorrect="off"
      id="applicationEmail"
      name="applicationEmail"
      placeholder="johnsmith123@email.com"
      type="email"
    />
    <br />
    <label
      htmlFor="applicationPosition"
    >
      Interested Position(s):
    </label>
    <br />
    <select
      id="applicationPosition"
      name="applicationPosition"
    >
      <option
        value="Spanish teacher"
      >
        Spanish teacher
      </option>
      <option
        value="French teacher"
      >
        French teacher
      </option>
      <option
        value="Robotics instructor"
      >
        Robotics instructor
      </option>
      <option
        value="Art instructor"
      >
        Art instructor
      </option>
    </select>
    <br />
    <label
      className="required"
      htmlFor="applicationResume"
    >
      Resume(Link)
    </label>
    <br />
    <input
      autoCapitalize="none"
      autoCorrect="off"
      id="applicationResume"
      name="applicationResume"
      placeholder="https://drive.google.com/file/d/examplelink3648384748/view"
      required={true}
      type="url"
    />
    <br />
    <button
      id="application-submit-btn"
      type="submit"
    >
      <b>
        Apply
      </b>
    </button>
  </form>
</div>
`;
